src/template/mod.rs

```diff
@@ -0,0 +1,198 @@
+mod block;
+mod parse;
+mod span;
+
+use color_eyre::eyre::{eyre, Result};
+
+use self::block::{Block, BlockKind, If, Var, VarEnv};
+use self::parse::Parser;
+use self::span::Spanned;
+use crate::variables::{UserVars, Variables};
+
+// TODO: handle unicode
```

Comment:
please remove todos and open issues instead before merging or committing to master

Reply:
This is/will be done on the branch `cleanup`